src/cpp-simulator/qiskitsimulator.py

```diff
@@ -0,0 +1,184 @@
+# =============================================================================
+
+"""
+Interface to Realistic Noise Simulator.
+
+Date: Nov 29 2017
+Author: Christopher J. Wood <cjwood@us.ibm.com>
+"""
+
+import os
+import json
+import subprocess
+from subprocess import PIPE, CalledProcessError
+import numpy as np
+import numbers
+
+
+def simulator_config():
+    """
+    Return the configuration dictionary for local_qiskit_simulator.
+    """
+    return {
+        "name": "local_qiskit_simulator",
+        "url": "https://github.com/QISKit/qiskit-sdk-py/external/cpp-simulator",
+        "simulator": True,
+        'local': True,
+        "description": "A C++ realistic noise simulator for qobj files",
+        "coupling_map": "all-to-all",
+        "basis_gates": "u1,u2,u3,cx,id,x,y,z,h,s,sdg,t,tdg,wait,noise,save,load,uzz"
+    }
+
+
+def clifford_config():
+    """
+    Return the configuration dictionary for local_clifford_simulator.
+    """
+    return {
+        "name": "local_clifford_simulator",
+        "url": "https://github.com/QISKit/qiskit-sdk-py/external/cpp-simulator",
+        "simulator": True,
+        'local': True,
+        "description": "A C++ Clifford simulator with approximate noise",
+        "coupling_map": "all-to-all",
+        "basis_gates": "cx,id,x,y,z,h,s,sdg,wait,noise,save,load"
+    }
+
+
+def run(qobj, exe='local_qiskit_simulator', path=None):
+    """
+    Run simulation on C++ simulator.
+
+    Args:
+        qobj (dict): qobj dictionary defining the simulation to run
+        exe (str): qobj simulator executable (default: 'wood_simulator')
+        path (str): path to exe directory (default: None)
+
+    Returns:
+        A dict of simulation results
+    """
+    # Get path to exe
+    if path is None:
+        cmd = os.path.dirname(os.path.abspath(__file__))
+    else:
+        cmd = os.path.expanduser(path)
+    # Add exe name
+    cmd = os.path.join(cmd, exe)
+
+    if 'config' in qobj:
+        qobj['config'] = __to_json_complex(qobj['config'])
+    for j in range(len(qobj['circuits'])):
+        if 'config' in qobj['circuits'][j]:
+            qobj['circuits'][j]['config'] = __to_json_complex(
+                qobj['circuits'][j]['config'])
+
+    # Open subprocess and execute external command
+    try:
+        with subprocess.Popen([cmd, '-'],
+                              stdin=PIPE, stdout=PIPE, stderr=PIPE) as proc:
+            cin = json.dumps(qobj).encode()
+            cout, cerr = proc.communicate(cin)
+        if len(cerr) > 0:
+            print('Simulator encountered a runtime error:')
+            print(cerr.decode())
+
+        cresult = json.loads(cout.decode())
+
+        if 'result' in cresult:
+            for result in cresult['result']:
+                if result['success'] is True:
+                    __parse_sim_data(result['data'])
+        return cresult
+
+    except FileNotFoundError:
+        msg = "ERROR: Simulator exe not found at: " + cmd
+        print(">> " + msg)
```

Comment:
A very good reason must exists in order to use `print` in the SDK... we banned it in favor of our logger.

Reply:
Moved this to the logger